### PR TITLE
[/tp command] - Check if x/y/z is valid

### PR DESCRIPTION
**Pull request by andriycraft**

### CODE REVIEW

File: src/lib/plugins/tp.js [hunk 1]
```diff
@@ -25,6 +25,11 @@ module.exports.server = (serv) => {
         let y = serv.posFromString(args[1], ctx.player.position.y)
         let z = serv.posFromString(args[2], ctx.player.position.z)
 
+        if (x > 29999999 || y > 4096 || y < -4096 || z > 29999999) {
```
Comment (darksunlight) [inline src/lib/plugins/tp.js:28]: What about negative x and z coordinates? You can check for the absolute values of x y and z instead.

**Review by darksunlight — CHANGES_REQUESTED** [review]

File: src/lib/plugins/tp.js [hunk 2]
```diff
@@ -25,6 +25,11 @@ module.exports.server = (serv) => {
         let y = serv.posFromString(args[1], ctx.player.position.y)
         let z = serv.posFromString(args[2], ctx.player.position.z)
 
+        if (x > 29999999 || y > 4096 || y < -4096 || z > 29999999 || z < -29999999 || x < -29999999) {
```
Comment (MokiyCodes) [inline src/lib/plugins/tp.js:28]: using `Math.abs(x) > 29999999 || Math.abs(y) > 4096 || Math.abs(z) > 29999999` would likely look nicer

**Review by MokiyCodes — CHANGES_REQUESTED** [review]
using `Math.abs(x) > 29999999 || Math.abs(y) > 4096 || Math.abs(z) > 29999999` would likely look nicer

**Comment by andriycraft** [comment]
> using `Math.abs(x) > 29999999 || Math.abs(y) > 4096 || Math.abs(z) > 29999999` would likely look nicer

Remember that we also have for negative x/y/z
so your code will look like this
`Math.abs(x) > 29999999 || Math.abs(y) > 4096 || Math.abs(z) > 29999999 || Math.abs(x) < -29999999 || Math.abs(y) < -4096 || Math.abs(z) < -29999999`

so I think my code is nicer

**Review by andriy332 — APPROVED** [review]

**Comment by darksunlight** [comment]
> > using `Math.abs(x) > 29999999 || Math.abs(y) > 4096 || Math.abs(z) > 29999999` would likely look nicer
> 
> Remember that we also must have check for negative x/y/z so your code will look like this `Math.abs(x) > 29999999 || Math.abs(y) > 4096 || Math.abs(z) > 29999999 || Math.abs(x) < -29999999 || Math.abs(y) < -4096 || Math.abs(z) < -29999999`
> 
> so I think my code is nicer

No. Not really. Math.abs returns the absolute value of its first parameter. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs

**Review by MokiyCodes — APPROVED** [review]
lgtm

**Pull request merged**